spirv_msl.cpp

```diff
@@ -1336,6 +1394,8 @@ uint32_t CompilerMSL::get_metal_resource_index(SPIRVariable &var, SPIRType::Base
 	switch (basetype)
 	{
 	case SPIRType::Struct:
+		if (execution.model == ExecutionModelVertex && next_metal_resource_index.msl_buffer == 0)
+			next_metal_resource_index.msl_buffer = 1;
```

Comment:
When i did this, i did not saw that i can pass `MSLResourceBinding` to set buffer index. I removed this two lines.